dev/generate_requirements.py

```diff
@@ -0,0 +1,112 @@
+"""
+This script generates a pip-compatible requirements file (e.g. requirements/core-requirements.txt)
+based on a YAML dependencies specification (e.g. requirements/core-requirements.yaml). These
+requirements files are then read by `setup.py` to specify dependencies for MLflow wheel
+installation.
+"""
```

Comment:
I think it's best to generate requirements.txt files from YAML, rather than having `setup.py` parse YAML directly, because: 1. pyyaml is required to parse yaml and 2. It's easier for developers / users to view the required dependencies in a requirements.txt file rather than reading / interpreting our custom yaml specifications.

Reply:
sounds good :)